d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -201,16 +210,17 @@ public void getClient(Request request, RequestContext requestContext, Callback<T
         }
         else
         {
-          _log.debug("service hint found, using generic client for target: {}", targetService);
-
-          TransportClient transportClient = _state.getClient(serviceName, targetService.getScheme());
+          URI target = override == null ? targetService : URI.create(override + service.getPath());
+          _log.debug("Host override found, using generic client for target: {}", target);
+          TransportClient transportClient = _state.getClient(serviceName, target.getScheme());
           if (transportClient == null)
           {
             throw new ServiceUnavailableException(serviceName,
-                "PEGA_1001. Cannot find transportClient for service " + serviceName + " and scheme: " + targetService.getScheme()
-                    + " with service hint" + targetService);
+                "PEGA_1001. Cannot find transportClient for service " + serviceName + " and scheme: " + target.getScheme()
+                    + " with host override " + target);
```

Comment:
Ditto - should we include TargetService in the error until we completely remove the TargetHost?